XMLParser/XMLParser.cpp

```diff
@@ -0,0 +1,361 @@
+// This is an open source non-commercial project. Dear PVS-Studio, please check it.
+
+// PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
+#include "pch.h"
+#include "framework.h"
+#include "XMLParser.h"
+
+bool XMLParser::ParseFile(const std::string& filename)
+{
+	struct Characters
+	{
+		const char tag1 = '<';
+		const char tag2 = '>';
+		const char slash = '/';
+		const std::string exclamation = "<!";
+		const std::string question = "<?";
+		const std::string end_tag_symbols = "</";
+	};
+	std::filesystem::path file_path = filename;
+	if (!FileCheck(file_path))
+	{
+		return false;
+	}
+
+	std::ifstream file{ filename };
+	if (!file.is_open())
+	{
+		//std::cout << "Can't open file.\n";
```

Comment:
Don't forget to change it for logger functions.

Reply:
Yes, sir.